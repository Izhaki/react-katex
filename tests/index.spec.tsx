import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import snapshotDiff from 'snapshot-diff';
import { ParseError } from 'katex';

import TeX from '../src';

const _ = String.raw;

describe('TeX', () => {
  test('expression via props.children', () => {
    expect(render(<TeX>\sum_0^\infty</TeX>).container).toMatchSnapshot();
  });

  test('expression via props.math', () => {
    expect(render(<TeX math={_`\sum_0^\infty`} />).container).toMatchSnapshot();
  });

  test('inline math by default', () => {
    const $ = render(<TeX data-testid="wrapper" math={_`\sum_0^\infty`} />);
    expect($.getByTestId('wrapper').tagName).toBe('SPAN');
  });

  test('block math via props.block', () => {
    const $ = render(
      <TeX data-testid="wrapper" math={_`\sum_0^\infty`} block />
    );
    expect($.getByTestId('wrapper').tagName).toBe('DIV');
  });

  test('customizing element type via props.as', () => {
    const $ = render(
      <TeX data-testid="wrapper" as="var">
        \sum_0^\infty
      </TeX>
    );
    expect($.getByTestId('wrapper').tagName).toBe('VAR');
    expect($.container).toMatchSnapshot();
  });

  test('customizing block element type via props.as & props.block', () => {
    const $ = render(
      <TeX data-testid="wrapper" as="figcaption" block>
        \sum_0^\infty
      </TeX>
    );
    expect($.getByTestId('wrapper').tagName).toBe('FIGCAPTION');
    expect($.container).toMatchSnapshot();
  });

  test('configuring KaTeX via props.settings', () => {
    const $ = render(
      <TeX settings={{ macros: { '*': 'cdot' } }}>y = k * x + c</TeX>
    );
    expect($.container).toMatchSnapshot();
  });

  test('props updated', () => {
    const $ = render(<TeX>\sum_0^\infty</TeX>);
    const $first = $.container.cloneNode(true);
    let $second;

    $.rerender(<TeX>\sum_1^\infty</TeX>);
    $second = $.container.cloneNode(true);

    expect(snapshotDiff($first, $second)).toMatchSnapshot();
  });

  describe('error handling', () => {
    test('show error when it updates from invalid to valid formula', () => {
      const renderError = (error: ParseError | TypeError) => (
        <p>{error.message} Cannot render this formula.</p>
      );
      const $ = render(<TeX renderError={renderError}>\inta</TeX>);
      expect($.queryByText(/Cannot render this formula/)).not.toBeNull();

      $.rerender(<TeX renderError={renderError}>\sum_1^\infty</TeX>);
      expect($.queryByText(/Cannot render this formula/)).toBeNull();
    });

    test('show error when it updates from invalid to valid formula', () => {
      const renderError = (error: ParseError | TypeError) => (
        <p>{error.message} Cannot render this formula.</p>
      );
      const $ = render(<TeX renderError={renderError}>\sum_1^\infty</TeX>);
      expect($.queryByText(/Cannot render this formula/)).toBeNull();

      $.rerender(<TeX renderError={renderError}>\inta</TeX>);
      expect($.queryByText(/Cannot render this formula/)).not.toBeNull();
    });

    test('formula with the wrong part highlighted in default color', () => {
      const $ = render(<TeX>\inta</TeX>);
      expect($.container).toMatchSnapshot();
    });

    test('props.errorColor', () => {
      const $ = render(<TeX errorColor="#bb0000">\inta</TeX>);
      expect($.container).toMatchSnapshot();
    });

    test('invalid type of expression in props.math', () => {
      let $;
      /* eslint-disable-next-line */
      console.error = jest.fn();
      $ = render(<TeX math={123} />);
      expect(
        $.queryByText(/KaTeX can only parse string typed expression/)
      ).not.toBeNull();
    });

    test('error is caused while parsing math expression', () => {
      let $;
      $ = render(<TeX math={_`\sum_{`} />);
      expect($.container).toMatchSnapshot();
    });

    test('props.renderError: invalid type of expression in props.math', () => {
      const renderError = (error: ParseError | TypeError) => (
        <div>Message of error: {error.message}</div>
      );
      let $;
      $ = render(<TeX math={123} renderError={renderError} />);
      expect($.container).toMatchSnapshot();
    });

    test('props.renderError: error is caused while parsing math expression', () => {
      const renderError = (error: ParseError | TypeError) => (
        <div>Message of error: {error.message}</div>
      );
      let $;
      $ = render(<TeX math={_`\sum_{`} renderError={renderError} />);
      expect($.container).toMatchSnapshot();
    });
  });

  describe('passing through other props', () => {
    test('style', () => {
      const $ = render(<TeX style={{ background: 'blue' }}>1 + 2 = 3</TeX>);
      expect($.container).toMatchSnapshot();
    });

    test('user style overwriting display style added by props.as', () => {
      const $ = render(
        <TeX style={{ display: 'inline' }} as="figcaption" block>
          1 + 2 = 3
        </TeX>
      );
      expect($.container).toMatchSnapshot();
    });

    test('className', () => {
      const $ = render(<TeX className="foo">1 + 2 = 3</TeX>);
      expect($.container).toMatchSnapshot();
    });
  });
});
