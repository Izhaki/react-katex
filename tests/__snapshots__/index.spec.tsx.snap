// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeX configuring KaTeX via props.settings 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <mi>
                y
              </mi>
              <mo>
                =
              </mo>
              <mi>
                k
              </mi>
              <mi>
                c
              </mi>
              <mi>
                d
              </mi>
              <mi>
                o
              </mi>
              <mi>
                t
              </mi>
              <mi>
                x
              </mi>
              <mo>
                +
              </mo>
              <mi>
                c
              </mi>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              y = k * x + c
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.625em;vertical-align:-0.19444em;"
          />
          <span
            class="mord mathnormal"
            style="margin-right:0.03588em;"
          >
            y
          </span>
          <span
            class="mspace"
            style="margin-right:0.2777777777777778em;"
          />
          <span
            class="mrel"
          >
            =
          </span>
          <span
            class="mspace"
            style="margin-right:0.2777777777777778em;"
          />
        </span>
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.77777em;vertical-align:-0.08333em;"
          />
          <span
            class="mord mathnormal"
            style="margin-right:0.03148em;"
          >
            k
          </span>
          <span
            class="mord mathnormal"
          >
            c
          </span>
          <span
            class="mord mathnormal"
          >
            d
          </span>
          <span
            class="mord mathnormal"
          >
            o
          </span>
          <span
            class="mord mathnormal"
          >
            t
          </span>
          <span
            class="mord mathnormal"
          >
            x
          </span>
          <span
            class="mspace"
            style="margin-right:0.2222222222222222em;"
          />
          <span
            class="mbin"
          >
            +
          </span>
          <span
            class="mspace"
            style="margin-right:0.2222222222222222em;"
          />
        </span>
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.43056em;vertical-align:0em;"
          />
          <span
            class="mord mathnormal"
          >
            c
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX customizing block element type via props.as & props.block 1`] = `
<div>
  <figcaption
    data-testid="wrapper"
  >
    <span
      class="katex-display"
    >
      <span
        class="katex"
      >
        <span
          class="katex-mathml"
        >
          <math
            display="block"
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <semantics>
              <mrow>
                <munderover>
                  <mo>
                    ∑
                  </mo>
                  <mn>
                    0
                  </mn>
                  <mi
                    mathvariant="normal"
                  >
                    ∞
                  </mi>
                </munderover>
              </mrow>
              <annotation
                encoding="application/x-tex"
              >
                \\sum_0^\\infty
              </annotation>
            </semantics>
          </math>
        </span>
        <span
          aria-hidden="true"
          class="katex-html"
        >
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:2.9185100000000004em;vertical-align:-1.267113em;"
            />
            <span
              class="mop op-limits"
            >
              <span
                class="vlist-t vlist-t2"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:1.6513970000000002em;"
                  >
                    <span
                      style="top:-1.882887em;margin-left:0em;"
                    >
                      <span
                        class="pstrut"
                        style="height:3.05em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          0
                        </span>
                      </span>
                    </span>
                    <span
                      style="top:-3.050005em;"
                    >
                      <span
                        class="pstrut"
                        style="height:3.05em;"
                      />
                      <span>
                        <span
                          class="mop op-symbol large-op"
                        >
                          ∑
                        </span>
                      </span>
                    </span>
                    <span
                      style="top:-4.3000050000000005em;margin-left:0em;"
                    >
                      <span
                        class="pstrut"
                        style="height:3.05em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          ∞
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="vlist-s"
                  >
                    ​
                  </span>
                </span>
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:1.267113em;"
                  >
                    <span />
                  </span>
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </span>
  </figcaption>
</div>
`;

exports[`TeX customizing element type via props.as 1`] = `
<div>
  <var
    data-testid="wrapper"
  >
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <msubsup>
                <mo>
                  ∑
                </mo>
                <mn>
                  0
                </mn>
                <mi
                  mathvariant="normal"
                >
                  ∞
                </mi>
              </msubsup>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\sum_0^\\infty
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:1.104002em;vertical-align:-0.29971000000000003em;"
          />
          <span
            class="mop"
          >
            <span
              class="mop op-symbol small-op"
              style="position:relative;top:-0.0000050000000000050004em;"
            >
              ∑
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t vlist-t2"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:0.804292em;"
                  >
                    <span
                      style="top:-2.40029em;margin-left:0em;margin-right:0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height:2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          0
                        </span>
                      </span>
                    </span>
                    <span
                      style="top:-3.2029em;margin-right:0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height:2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          ∞
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="vlist-s"
                  >
                    ​
                  </span>
                </span>
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:0.29971000000000003em;"
                  >
                    <span />
                  </span>
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </span>
  </var>
</div>
`;

exports[`TeX error handling error is caused while parsing math expression 1`] = `
<div>
  <span>
    <span
      class="katex-error"
      style="color:#cc0000"
      title="ParseError: KaTeX parse error: Expected '}', got 'EOF' at end of input: \\\\sum_{"
    >
      \\sum_{
    </span>
  </span>
</div>
`;

exports[`TeX error handling formula with the wrong part highlighted in default color 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <mstyle
                mathcolor="#cc0000"
              >
                <mtext>
                  \\inta
                </mtext>
              </mstyle>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\inta
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:1em;vertical-align:-0.25em;"
          />
          <span
            class="mord text"
            style="color:#cc0000;"
          >
            <span
              class="mord"
              style="color:#cc0000;"
            >
              \\inta
            </span>
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX error handling props.errorColor 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <mstyle
                mathcolor="#bb0000"
              >
                <mtext>
                  \\inta
                </mtext>
              </mstyle>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\inta
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:1em;vertical-align:-0.25em;"
          />
          <span
            class="mord text"
            style="color:#bb0000;"
          >
            <span
              class="mord"
              style="color:#bb0000;"
            >
              \\inta
            </span>
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX error handling props.renderError: error is caused while parsing math expression 1`] = `
<div>
  <div>
    Message of error: 
    KaTeX parse error: Expected '}', got 'EOF' at end of input: \\sum_{
  </div>
</div>
`;

exports[`TeX error handling props.renderError: invalid type of expression in props.math 1`] = `
<div>
  <div>
    Message of error: 
    KaTeX can only parse string typed expression
  </div>
</div>
`;

exports[`TeX expression via props.children 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <msubsup>
                <mo>
                  ∑
                </mo>
                <mn>
                  0
                </mn>
                <mi
                  mathvariant="normal"
                >
                  ∞
                </mi>
              </msubsup>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\sum_0^\\infty
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:1.104002em;vertical-align:-0.29971000000000003em;"
          />
          <span
            class="mop"
          >
            <span
              class="mop op-symbol small-op"
              style="position:relative;top:-0.0000050000000000050004em;"
            >
              ∑
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t vlist-t2"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:0.804292em;"
                  >
                    <span
                      style="top:-2.40029em;margin-left:0em;margin-right:0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height:2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          0
                        </span>
                      </span>
                    </span>
                    <span
                      style="top:-3.2029em;margin-right:0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height:2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          ∞
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="vlist-s"
                  >
                    ​
                  </span>
                </span>
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:0.29971000000000003em;"
                  >
                    <span />
                  </span>
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX expression via props.math 1`] = `
<div>
  <span>
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <msubsup>
                <mo>
                  ∑
                </mo>
                <mn>
                  0
                </mn>
                <mi
                  mathvariant="normal"
                >
                  ∞
                </mi>
              </msubsup>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\sum_0^\\infty
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:1.104002em;vertical-align:-0.29971000000000003em;"
          />
          <span
            class="mop"
          >
            <span
              class="mop op-symbol small-op"
              style="position:relative;top:-0.0000050000000000050004em;"
            >
              ∑
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t vlist-t2"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:0.804292em;"
                  >
                    <span
                      style="top:-2.40029em;margin-left:0em;margin-right:0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height:2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          0
                        </span>
                      </span>
                    </span>
                    <span
                      style="top:-3.2029em;margin-right:0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height:2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          ∞
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="vlist-s"
                  >
                    ​
                  </span>
                </span>
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height:0.29971000000000003em;"
                  >
                    <span />
                  </span>
                </span>
              </span>
            </span>
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX passing through other props className 1`] = `
<div>
  <span
    class="foo"
  >
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <mn>
                1
              </mn>
              <mo>
                +
              </mo>
              <mn>
                2
              </mn>
              <mo>
                =
              </mo>
              <mn>
                3
              </mn>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              1 + 2 = 3
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.72777em;vertical-align:-0.08333em;"
          />
          <span
            class="mord"
          >
            1
          </span>
          <span
            class="mspace"
            style="margin-right:0.2222222222222222em;"
          />
          <span
            class="mbin"
          >
            +
          </span>
          <span
            class="mspace"
            style="margin-right:0.2222222222222222em;"
          />
        </span>
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.64444em;vertical-align:0em;"
          />
          <span
            class="mord"
          >
            2
          </span>
          <span
            class="mspace"
            style="margin-right:0.2777777777777778em;"
          />
          <span
            class="mrel"
          >
            =
          </span>
          <span
            class="mspace"
            style="margin-right:0.2777777777777778em;"
          />
        </span>
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.64444em;vertical-align:0em;"
          />
          <span
            class="mord"
          >
            3
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX passing through other props style 1`] = `
<div>
  <span
    style="background: blue;"
  >
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <mn>
                1
              </mn>
              <mo>
                +
              </mo>
              <mn>
                2
              </mn>
              <mo>
                =
              </mo>
              <mn>
                3
              </mn>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              1 + 2 = 3
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.72777em;vertical-align:-0.08333em;"
          />
          <span
            class="mord"
          >
            1
          </span>
          <span
            class="mspace"
            style="margin-right:0.2222222222222222em;"
          />
          <span
            class="mbin"
          >
            +
          </span>
          <span
            class="mspace"
            style="margin-right:0.2222222222222222em;"
          />
        </span>
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.64444em;vertical-align:0em;"
          />
          <span
            class="mord"
          >
            2
          </span>
          <span
            class="mspace"
            style="margin-right:0.2777777777777778em;"
          />
          <span
            class="mrel"
          >
            =
          </span>
          <span
            class="mspace"
            style="margin-right:0.2777777777777778em;"
          />
        </span>
        <span
          class="base"
        >
          <span
            class="strut"
            style="height:0.64444em;vertical-align:0em;"
          />
          <span
            class="mord"
          >
            3
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`TeX passing through other props user style overwriting display style added by props.as 1`] = `
<div>
  <figcaption
    style="display: inline;"
  >
    <span
      class="katex-display"
    >
      <span
        class="katex"
      >
        <span
          class="katex-mathml"
        >
          <math
            display="block"
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <semantics>
              <mrow>
                <mn>
                  1
                </mn>
                <mo>
                  +
                </mo>
                <mn>
                  2
                </mn>
                <mo>
                  =
                </mo>
                <mn>
                  3
                </mn>
              </mrow>
              <annotation
                encoding="application/x-tex"
              >
                1 + 2 = 3
              </annotation>
            </semantics>
          </math>
        </span>
        <span
          aria-hidden="true"
          class="katex-html"
        >
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:0.72777em;vertical-align:-0.08333em;"
            />
            <span
              class="mord"
            >
              1
            </span>
            <span
              class="mspace"
              style="margin-right:0.2222222222222222em;"
            />
            <span
              class="mbin"
            >
              +
            </span>
            <span
              class="mspace"
              style="margin-right:0.2222222222222222em;"
            />
          </span>
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:0.64444em;vertical-align:0em;"
            />
            <span
              class="mord"
            >
              2
            </span>
            <span
              class="mspace"
              style="margin-right:0.2777777777777778em;"
            />
            <span
              class="mrel"
            >
              =
            </span>
            <span
              class="mspace"
              style="margin-right:0.2777777777777778em;"
            />
          </span>
          <span
            class="base"
          >
            <span
              class="strut"
              style="height:0.64444em;vertical-align:0em;"
            />
            <span
              class="mord"
            >
              3
            </span>
          </span>
        </span>
      </span>
    </span>
  </figcaption>
</div>
`;

exports[`TeX props updated 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -14,11 +14,11 @@
                <msubsup>
                  <mo>
                    ∑
                  </mo>
                  <mn>
-                   0
+                   1
                  </mn>
                  <mi
                    mathvariant=\\"normal\\"
                  >
                    ∞
@@ -26,11 +26,11 @@
                </msubsup>
              </mrow>
              <annotation
                encoding=\\"application/x-tex\\"
              >
-               \\\\sum_0^\\\\infty
+               \\\\sum_1^\\\\infty
              </annotation>
            </semantics>
          </math>
        </span>
        <span
@@ -77,11 +77,11 @@
                          class=\\"sizing reset-size6 size3 mtight\\"
                        >
                          <span
                            class=\\"mord mtight\\"
                          >
-                           0
+                           1
                          </span>
                        </span>
                      </span>
                      <span
                        style=\\"top:-3.2029em;margin-right:0.05em;\\""
`;
